Reemit.Common/RangeMappedComparer.cs

```diff
@@ -0,0 +1,24 @@
+﻿using System.Diagnostics.CodeAnalysis;
+
+namespace Reemit.Common;
+
+public class RangeMappedComparer : IEqualityComparer<IRangeMapped>
+{
+    public static RangeMappedComparer Default { get; } = new RangeMappedComparer();    
+
+    public bool Equals(IRangeMapped? x, IRangeMapped? y) =>
+        x == null && y == null ?
+            true :
+        x == null || y == null ?
+            false :
+            x.Position == y.Position && x.Length == y.Length;
+
+    public int GetHashCode([DisallowNull] IRangeMapped obj)
```

Comment:
Rider says we can delete the DisallowNull attribute here.
<img width="657" alt="image" src="https://github.com/v0idzz/Reemit/assets/12448522/8ed728af-45d1-4126-badd-670de3c0b064">

Reply:
Interesting, that does appear to be the case since IRangeMappeds isn't nullable. I used the VS implement interface feature to generate the initial method declarations here. Seems there might be some room for improvement in it. Done.